src/interpreter.rs

```diff
@@ -1,222 +1,432 @@
-use crate::ast::{BinaryOperator, Expression, LogicalExpression, Value};
+use crate::ast::{self, BinaryOperator, Expression, LogicalExpression, Type, Value};
 use crate::context::{Context, Match};
+use crate::schema;
 
 pub trait Execute {
-    fn execute(&self, context: &mut Context, m: &mut Match) -> bool;
+    fn execute(&self, ctx: &mut Context, m: &mut Match) -> bool;
 }
 
 impl Execute for Expression {
-    fn execute(&self, context: &mut Context, m: &mut Match) -> bool {
+    fn execute(&self, ctx: &mut Context, m: &mut Match) -> bool {
         match self {
             Expression::Logical(l) => match l.as_ref() {
-                LogicalExpression::And(l, r) => l.execute(context, m) && r.execute(context, m),
-                LogicalExpression::Or(l, r) => l.execute(context, m) || r.execute(context, m),
+                LogicalExpression::And(l, r) => l.execute(ctx, m) && r.execute(ctx, m),
+                LogicalExpression::Or(l, r) => l.execute(ctx, m) || r.execute(ctx, m),
             },
-            Expression::Predicate(p) => {
-                let lhs_values = match context.value_of(&p.lhs.var_name) {
-                    None => return false,
-                    Some(v) => v,
-                };
-
-                let (lower, any) = p.lhs.get_transformations();
-
-                for mut lhs_value in lhs_values
-                    .iter()
-                    .take(if any { lhs_values.len() } else { 1 })
-                {
-                    let lhs_value_transformed;
-
-                    if lower {
-                        match lhs_value {
-                            Value::String(s) => {
-                                lhs_value_transformed = Value::String(s.to_lowercase());
-                                lhs_value = &lhs_value_transformed;
-                            }
-                            _ => unreachable!(),
+            Expression::Predicate(p) => p.execute(ctx, m),
+        }
+    }
+}
+
+impl Execute for ast::Predicate {
```

Comment:
```suggestion
impl Execute for Predicate {
```

Let's import `Predicate` directly to stay consistent with the style of this project.